jdaviz/configs/imviz/plugins/compass/compass.py

```diff
@@ -3,12 +3,25 @@
 from jdaviz.core.events import AddDataMessage, RemoveDataMessage
 from jdaviz.core.registries import tray_registry
 from jdaviz.core.template_mixin import PluginTemplateMixin, ViewerSelectMixin
+from jdaviz.core.user_api import PluginUserApi
 
 __all__ = ['Compass']
 
 
 @tray_registry('imviz-compass', label="Imviz Compass")
 class Compass(PluginTemplateMixin, ViewerSelectMixin):
+    """
+    See the :ref:`Compass Plugin Documentation <imviz-compass>` for more details.
+
+    Only the following attributes and methods are available through the
+    :ref:`public plugin API <plugin-apis>`:
+
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
+    * ``viewer`` (:class:`~jdaviz.core.template_mixin.ViewerSelect`):
+      Viewer to show orientation/compass information.
+    * ``data_label``: label of the top-layer shown in the compass (read-only)
+    """
     template_file = __file__, "compass.vue"
     icon = Unicode("").tag(sync=True)
     data_label = Unicode("").tag(sync=True)

```

Comment:
is `data_label` the most appropriate here (as opposed to `layer_label`, `top_layer`, etc)?  We probably will use similar patterns in other plugins, so might want to give this a little thought.

Reply:
I was thinking about that too; the reason why I said "or something like that" is because I feel like `data_label` isn't exactly the right wording, but it's the name of the traitlet. Personally I don't think its THAT horrible of a name, so I'm willing to let it slide

Reply:
we can rename internal traitlets at this point easily, but once we expose it to the user-API, that would likely count as API-breaking.

Reply:
Borrowing the concern from #1583, are there any other plugins that use the same traitlet name that we should be sensitive to?

Reply:
I don't think any other plugins currently use this, but I think there was a suggestion/request to have it in line profiles.  @pllim would know better since relying on the "top layer" logic is pretty imviz-specific.

Reply:
Yes, top layer is Imviz specific for now. But... anything that uses the coordinates info panel or any astrowidgets API and allow loading multiple datasets into the same viewer should support it.

Reply:
If you want to be even more specific, it is actually the top image layer that is 2D and not a Subset and also not set to invisible. Name that how you will. 😸 

https://github.com/spacetelescope/jdaviz/blob/78b5d3584784a3e9cf134b506935e9e79d872064/jdaviz/configs/imviz/helper.py#L415-L425

Reply:
That's a good point, it always will be an image (not another type of plotting layer/marker/etc), so maybe `data_label` is the most descriptive.